jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/audit/CaseRoleAssignmentLogDeleteBuilderImpl.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.casemgmt.impl.audit;
+
+import org.jbpm.process.audit.JPAAuditLogService;
+import org.jbpm.process.audit.query.AbstractAuditDeleteBuilderImpl;
+import org.kie.api.runtime.CommandExecutor;
+import org.kie.internal.query.QueryParameterIdentifiers;
+
+public class CaseRoleAssignmentLogDeleteBuilderImpl extends AbstractAuditDeleteBuilderImpl<CaseRoleAssignmentLogDeleteBuilderImpl> {
+
+    private final static String CASE_ROLE_ASSIGMENT_LOG_DELETE = "DELETE FROM CaseRoleAssignmentLog l";
+
+    protected String caseDefId;
+
+    public CaseRoleAssignmentLogDeleteBuilderImpl(JPAAuditLogService jpaService) {
+        super(jpaService);
+        intersect();
+    }
+
+    public CaseRoleAssignmentLogDeleteBuilderImpl(CommandExecutor cmdExecutor) {
+        super(cmdExecutor);
+        intersect();
+    }
+
+    @Override
+    protected boolean isSubquerySupported() {
+        return true;
+    }
+
+    @Override
+    protected Subquery getSubQuery() {
+        return new Subquery("l.caseId", "select spl.correlationKey FROM ProcessInstanceLog spl", 1);
+    }
+
+    @Override
+    protected Subquery applyParameters(Subquery subquery) {
+        Subquery query = super.applyParameters(subquery);
+        query.parameter(QueryParameterIdentifiers.SUBQUERY_CASE, caseDefId);
+        return query;
+    }
+
+    public CaseRoleAssignmentLogDeleteBuilderImpl inCaseDefId(String caseDefId) {
```

Comment:
Still holds.

Reply:
@MarianMacik point out where in the spec we agreed is this.

Reply:
@mareknovotny ^^

Reply:
@MarianMacik if you think this is important you should add it into the document right now. Just as a side watcher of this PR review, we should be also aware of time we are spending on this PR review and if you have not major additional ideas how to improve/enhance solution, it would be sometime good to have it done later as a follow up. File a new jira related to it and let it done in small unit of work there.